src/ballet/sbpf/test_sbpf_opcode.c

```diff
@@ -0,0 +1,130 @@
+  FD_TEST( 0x54==BPF_OP_AND32_IMM  );
+  FD_TEST( 0x5c==BPF_OP_AND32_REG  );
+  FD_TEST( 0x64==BPF_OP_LSH32_IMM  );
+  FD_TEST( 0x6c==BPF_OP_LSH32_REG  );
+  FD_TEST( 0x74==BPF_OP_RSH32_IMM  );
+  FD_TEST( 0x7c==BPF_OP_RSH32_REG  );
+  FD_TEST( 0x84==BPF_OP_NEG32      );
+  FD_TEST( 0x94==BPF_OP_MOD32_IMM  );
+  FD_TEST( 0x9c==BPF_OP_MOD32_REG  );
+  FD_TEST( 0xa4==BPF_OP_XOR32_IMM  );
+  FD_TEST( 0xac==BPF_OP_XOR32_REG  );
+  FD_TEST( 0xb4==BPF_OP_MOV32_IMM  );
+  FD_TEST( 0xbc==BPF_OP_MOV32_REG  );
+  FD_TEST( 0xc4==BPF_OP_ARSH32_IMM );
+  FD_TEST( 0xcc==BPF_OP_ARSH32_REG );
+  FD_TEST( 0xe4==BPF_OP_SDIV32_IMM );
+  FD_TEST( 0xec==BPF_OP_SDIV32_REG );
+
+  /* 64-bit arithmetic opcodes */
+
+  FD_TEST( 0x07==BPF_OP_ADD64_IMM  );
+  FD_TEST( 0x0f==BPF_OP_ADD64_REG  );
+  FD_TEST( 0x17==BPF_OP_SUB64_IMM  );
+  FD_TEST( 0x1f==BPF_OP_SUB64_REG  );
+  FD_TEST( 0x27==BPF_OP_MUL64_IMM  );
+  FD_TEST( 0x2f==BPF_OP_MUL64_REG  );
+  FD_TEST( 0x37==BPF_OP_DIV64_IMM  );
+  FD_TEST( 0x3f==BPF_OP_DIV64_REG  );
+  FD_TEST( 0x47==BPF_OP_OR64_IMM   );
+  FD_TEST( 0x4f==BPF_OP_OR64_REG   );
+  FD_TEST( 0x57==BPF_OP_AND64_IMM  );
+  FD_TEST( 0x5f==BPF_OP_AND64_REG  );
+  FD_TEST( 0x67==BPF_OP_LSH64_IMM  );
+  FD_TEST( 0x6f==BPF_OP_LSH64_REG  );
+  FD_TEST( 0x77==BPF_OP_RSH64_IMM  );
+  FD_TEST( 0x7f==BPF_OP_RSH64_REG  );
+  FD_TEST( 0x87==BPF_OP_NEG64      );
+  FD_TEST( 0x97==BPF_OP_MOD64_IMM  );
+  FD_TEST( 0x9f==BPF_OP_MOD64_REG  );
+  FD_TEST( 0xa7==BPF_OP_XOR64_IMM  );
+  FD_TEST( 0xaf==BPF_OP_XOR64_REG  );
+  FD_TEST( 0xb7==BPF_OP_MOV64_IMM  );
+  FD_TEST( 0xbf==BPF_OP_MOV64_REG  );
+  FD_TEST( 0xc7==BPF_OP_ARSH64_IMM );
+  FD_TEST( 0xcf==BPF_OP_ARSH64_REG );
+  FD_TEST( 0xe7==BPF_OP_SDIV64_IMM );
+  FD_TEST( 0xef==BPF_OP_SDIV64_REG );
+
+  /* Control flow opcodes */
+
+  FD_TEST( 0x05==BPF_OP_JA         );
+  FD_TEST( 0x15==BPF_OP_JEQ_IMM    );
+  FD_TEST( 0x1d==BPF_OP_JEQ_REG    );
+  FD_TEST( 0x25==BPF_OP_JGT_IMM    );
+  FD_TEST( 0x2d==BPF_OP_JGT_REG    );
+  FD_TEST( 0x35==BPF_OP_JGE_IMM    );
+  FD_TEST( 0x3d==BPF_OP_JGE_REG    );
+  FD_TEST( 0x45==BPF_OP_JSET_IMM   );
+  FD_TEST( 0x4d==BPF_OP_JSET_REG   );
+  FD_TEST( 0x55==BPF_OP_JNE_IMM    );
+  FD_TEST( 0x5d==BPF_OP_JNE_REG    );
+  FD_TEST( 0x65==BPF_OP_JSGT_IMM   );
+  FD_TEST( 0x6d==BPF_OP_JSGT_REG   );
+  FD_TEST( 0x75==BPF_OP_JSGE_IMM   );
+  FD_TEST( 0x7d==BPF_OP_JSGE_REG   );
+  FD_TEST( 0xa5==BPF_OP_JLT_IMM    );
+  FD_TEST( 0xad==BPF_OP_JLT_REG    );
+  FD_TEST( 0xb5==BPF_OP_JLE_IMM    );
+  FD_TEST( 0xbd==BPF_OP_JLE_REG    );
+  FD_TEST( 0xc5==BPF_OP_JSLT_IMM   );
+  FD_TEST( 0xcd==BPF_OP_JSLT_REG   );
+  FD_TEST( 0xd5==BPF_OP_JSLE_IMM   );
+  FD_TEST( 0xdd==BPF_OP_JSLE_REG   );
+
+  /* Function call opcodes */
+
+  FD_TEST( 0x85==BPF_OP_CALL       );
+  FD_TEST( 0x8d==BPF_OP_CALLX      );
+  FD_TEST( 0x95==BPF_OP_EXIT       );
+
+  /* Miscellaneous opcodes */
+
+  FD_TEST( 0x18==BPF_OP_LDDW       );
+  FD_TEST( 0xd4==BPF_OP_LE         );
+  FD_TEST( 0xdc==BPF_OP_BE         );
```

Comment:
Probably would do all these FD_TESTS as a file scope FD_STATIC_ASSERT and the unit test itself is pretty much a default "pass" if the test compiles at all.